src/tokens.ts

```diff
@@ -16,26 +16,49 @@ import { IObservableDisposable } from '@phosphor/disposable';
 import { DebugProtocol } from 'vscode-debugprotocol';
 
 import { Debugger } from './debugger';
-
-import { DebuggerSidebar } from './sidebar';
+import { Session } from '@jupyterlab/services';
 
 /**
  * An interface describing an application's visual debugger.
  */
-export interface IDebugger extends IWidgetTracker<MainAreaWidget<Debugger>> {}
+export interface IDebugger {
+  /**
+   * The mode of the debugger UI.
+   *
+   * #### Notes
+   * There is only ever one debugger instance. If it is `expanded`, it exists
+   * as a `MainAreaWidget`, otherwise it is a sidebar.
+   */
+  mode: IDebugger.Mode;
+
+  /**
+   * The current debugger session.
+   */
+  session: IDebugger.ISession;
+
+  /**
+   * tracker for get instance of debugger.
+   */
+  tracker: WidgetTracker<MainAreaWidget<Debugger>>;
```

Comment:
Why do we need the tracker here? Shouldn't the interface only expose `mode` and `session`?

Reply:
For handlers in consoles and notebooks Plugin. they need access to the model of Debugger.
Should we create some different way to share model between plugins?

Reply:
Yeah ideally we would require the tracker in the plugin (with `requires`), but at the moment it's not provided anywhere.

Reply:
I see that it was changed in https://github.com/jupyterlab/debugger/pull/48 as well.